### PR TITLE
Ensure pip is installed when creating the virtual environment and many other fixes

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: cpp/lib.py [hunk 1]
```diff
@@ -865,7 +893,7 @@ class BBPProject:
         ClangFormat=dict(
             cls=ExecutableTool,
             name="clang-format",
-            names_glob_patterns=["clang-format-*"],
+            names_glob_patterns=["clang-format-[-a-z0-9]"],
```
Comment (olupton) [inline cpp/lib.py:896]: I'm not sure this is right, with `glob` isn't this just requiring a single character from the class?
Modifying the test to match this line (can this be enforced somehow?) would verify: https://github.com/BlueBrain/hpc-coding-conventions/blob/5879e6a7fd3bd1da5ab3b92ff86365663e8363ed/dev/test_lib.py#L29

**Review by olupton — COMMENTED** [review]
Generally looks good to me, one query about the pattern for clang-format.

File: cpp/lib.py (hunk 1 above)
Comment (tristan0x) [reply cpp/lib.py]: Thank you for catching this @olupton 
This is fixed in 44ceb89

**Review by olupton — APPROVED** [review]
LGTM 🚀

**Pull request merged**